clippy_lints/src/casts/cast_lossless.rs

```diff
@@ -77,7 +77,13 @@ pub(super) fn check(
 
 fn should_lint(cx: &LateContext<'_>, expr: &Expr<'_>, cast_from: Ty<'_>, cast_to: Ty<'_>, msrv: &Msrv) -> bool {
     // Do not suggest using From in consts/statics until it is valid to do so (see #2267).
-    if in_constant(cx, expr.hir_id) {
+    //
+    // If destination is (U)int128, do not lint because source type cannot be larger
+    // If source is bool, still lint due to the lint message differing (refers to style)
+    if in_constant(cx, expr.hir_id)
+        || (!cast_from.is_bool()
+            && (matches!(cast_to.kind(), ty::Int(IntTy::I128)) || matches!(cast_to.kind(), ty::Uint(UintTy::U128))))
```

Comment:
Casting from a  `u128` to a `i128` could be dangerous. `u128` is the only data type that we can be sure that they won't cause an error.

```suggestion
            && matches!(cast_to.kind(), ty::Uint(UintTy::U128))
```